### PR TITLE
CASMNET-1060 update templates for CMN net

### PR DESCRIPTION
**Pull request by lukebates123**
### Summary and Scope

Description: What does this change do?  Use examples of new options and output changes when possible.  If other changes were made list these as well in a list.

PR checklist (you may replace this section):
- [x] I have run `nox` locally and all tests, linting, and code coverage pass.
- [x] I have updated the appropriate Changelog entries in readme.md
- [x] I have added new tests to cover the new code
- [x] My code follows the style guidelines of this project
- [x] If adding a new file, I have updated pyinstaller.py

### Issues and Related PRs

CASMNET-1060

### Testing

Tested on:

Hela

### CODE REVIEW

File: canu/generate/switch/config/config.py [hunk 1]
```diff
@@ -565,7 +570,8 @@ def generate_switch_config(
     if switch_name not in sls_variables["HMN_IPs"].keys():
         click.secho(f"Cannot find {switch_name} in CSI / SLS nodes.", fg="red")
         exit(1)
-
+    if sls_variables["CMN_IPs"]:
```
Comment (trad511) [inline canu/generate/switch/config/config.py:573]: The way Python works checking this w/o it existing will blow up.  You could:
```
if "CMN_IPs" in sls_variables:
```
But a more Pythonic way to do this is:
```
myvar = sls_variables.get("CMN_IPs")
```
which returns None if the variable isn't set and you can test on that or...
```
variables["CMN_IP"] = sls_variables.get("CMN_IPs", whatever_you_want_default_to_be)
```

File: canu/generate/switch/config/config.py [hunk 2]
```diff
@@ -1097,6 +1103,8 @@ def groupby_vlan_range(vlan_list):
     Returns:
         list of vlans formatted
     """
+    if None in vlan_list:
```
Comment (trad511) [inline canu/generate/switch/config/config.py:1106]: Do you really want to throw out the entire list because of a single entry of "None"?  Wouldn't we just want to prune the list?  Not quite sure what you're doing here.  Also the return value is nothing and the docs for this function say returning a list of vlans.

File: canu/generate/switch/config/config.py [hunk 3]
```diff
@@ -1097,6 +1103,8 @@ def groupby_vlan_range(vlan_list):
...
+        return
     if not len(vlan_list):
```
Comment (trad511) [inline canu/generate/switch/config/config.py:1108]: I know this is an older bit of code, but this seems like a weird test result.  The doc says we return a (sorted) list of vlans, but here we return a singular string?  Probably need to make sure we're returning things consistently: a list, a string, something.

File: docs/generate_network_config.md [hunk 4]
```diff
@@ -69,7 +69,7 @@ canu generate network config [OPTIONS]
 
 
 ### -a(, --architecture( <architecture>)
-CSM architecture
+**Required** CSM architecture
```
Comment (trad511) [inline docs/generate_network_config.md:72]: Good idea

File: network_modeling/configs/templates/1.0/aruba/common/cdu-to-spine.lag.j2 [hunk 5]
```diff
@@ -7,8 +7,8 @@ interface lag 255 multi-chassis
     no shutdown
     description cdu_to_spines_lag
     no routing
-    vlan trunk native {{ variables.NATIVE_VLAN }}
-    vlan trunk allowed {{ variables.LEAF_BMC_VLANS }}
+    vlan trunk native 1
```
Comment (trad511) [inline network_modeling/configs/templates/1.0/aruba/common/cdu-to-spine.lag.j2:10]: More work required to make these dynamic?  Why this change?

Comment (lukebates123) [reply network_modeling/configs/templates/1.0/aruba/common/cdu-to-spine.lag.j2:10]: 1.0 templates, 1.2 templates are all dynamic.. the CMM vlan breaks 1.0.

File: canu/generate/switch/config/config.py (hunk 1 above)
Comment (lukebates123) [reply canu/generate/switch/config/config.py]: Fixed.

File: canu/generate/switch/config/config.py (hunk 2 above)
Comment (lukebates123) [reply canu/generate/switch/config/config.py]: Function now prunes out "None"

**Review by trad511 — APPROVED** [review]

**Review by dle-hpe — APPROVED** [review]
